lib/actions/apiV2.js

```diff
@@ -753,8 +765,30 @@ export function findRoutes() {
             // To initialize the route viewer,
             // convert the routes array to a dictionary indexed by route ids.
             return routes.reduce((result, route) => {
-              const { agency, color, id, longName, mode, shortName, type } =
-                route
+              const {
+                agency,
+                color: origColor,
+                id,
+                longName,
+                mode,
+                shortName,
+                type
+              } = route
+              // Set color overrides if present
+              const color = getRouteColorBasedOnSettings(
+                getRouteOperator(
+                  {
+                    agencyId: route?.agency?.id,
+                    id: route?.route?.id
+                  },
+                  config.transitOperators
+                ),
+                {
+                  color: route?.route?.color,
+                  mode: route.mode
+                }
+              ).split('#')?.[1]
```

Comment:
This method looks almost the same as the one above (and I think there's a third one in this file as well) is there a way to pull this out so you don't have to write it out twice?

Reply:
Yeah I thought about this but the problem is this is adapted from a core-utils method that's already slightly too wrong to work here so I was thinking you know let's just let this sit for a while, iron out all the changes and make sure that no tiny differences are needed and then we can clean this up in a future refactor!